src/librustc/ty/relate.rs

```diff
@@ -226,12 +226,16 @@ impl<'tcx> Relate<'tcx> for ty::ProjectionTy<'tcx> {
         where R: TypeRelation<'a, 'gcx, 'tcx>, 'gcx: 'a+'tcx, 'tcx: 'a
     {
         let tcx = relation.tcx();
-        if a.item_name(tcx) != b.item_name(tcx) {
+        if a.item_def_id != b.item_def_id {
+            // FIXME(tschottdorf): is this (still) the correct error?
```

Comment:
No, you want to rename `ProjectionNameMismatched` to `ProjectionMismatched` and store the `DefId`'s inside.